### PR TITLE
[Merged by Bors] - feat(Data/Finset/Sort): Every finite linear order embeds into every infinite linear order

### PR DESCRIPTION
**Pull request by awainverse**
Constructs `Fintype.orderIsoOfFin` along the lines of `Finset.orderIsoOfFin`
Shows that every finite linear order embeds into every infinite linear order

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by awainverse** [comment]
I'm not sure whether this is the right place for `nonempty_orderEmbedding_of_finite_infinite` - open to suggestions.

File: Mathlib/Data/Finset/Sort.lean [hunk 1]
```diff
@@ -241,3 +241,20 @@ theorem sort_univ (n : ℕ) : Finset.univ.sort (fun x y : Fin n => x ≤ y) = Li
     (List.pairwise_le_finRange n)
 
 end Fin
+
+/-- Given a `Fintype` `α` of cardinality `k`, the map `orderIsoOfFin s h` is the increasing
+bijection between `Fin k` and `α` as an `OrderIso`. Here, `h` is a proof that the cardinality of `α`
+is `k`. We use this instead of an iso `Fin (Fintype.card α) ≃o α` to avoid casting issues in further
+uses of this function. -/
+def Fintype.orderIsoOfFin (α : Type*) [LinearOrder α] [Fintype α] {k : ℕ} (h : Fintype.card α = k) :
```
Comment (YaelDillies) [inline Mathlib/Data/Finset/Sort.lean:249]:
```suggestion
def Fintype.orderIsoFinOfCardEq (α : Type*) [LinearOrder α] [Fintype α] {k : ℕ} (h : Fintype.card α = k) :
```
to match the `Equiv` version

**Review by YaelDillies — APPROVED** [review]
Thanks!

maintainer merge

**Comment by urkud** [comment]
Thanks! :tada:
bors merge

**Pull request closed without merging**